public/js/main.js

```diff
@@ -171,6 +170,19 @@ bingo.controller('homeController', function($scope, $http, $location, bingosocke
 
 bingo.controller('bingoController', function($scope, $document, $http, $routeParams, bingosockets) {
   // Responsive bingo card: keep squares square.
+
+  // Make sure that we warn the user before they leave the gameroom
+  $scope.$on('$locationChangeStart', function(event, next, current) {
+    if (1) {
+      var answer = confirm('Are you sure you want to leave the game room');
```

Comment:
I think this'll work for now. Ideally, we'd like an actual pop-up window. 
(Perhaps see https://material.angularjs.org/#/demo/material.components.dialog )